openedx/features/subscriptions/api/v1/views.py

```diff
@@ -0,0 +1,53 @@
+"""
+HTTP endpoints for interacting with subscriptions.
+"""
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.generics import ListAPIView, RetrieveUpdateAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from django.http import Http404
+
+from lms.djangoapps.commerce.api.v1.permissions import ApiKeyOrModelPermission
+from openedx.core.lib.api.mixins import PutAsCreateMixin
+from openedx.features.subscriptions.models import UserSubscription
+from openedx.features.subscriptions.api.v1.serializers import UserSubscriptionSerializer
+
+
+class SubscriptionsListView(ListAPIView):
+    """
+    List subscriptions.
+    """
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+    serializer_class = UserSubscriptionSerializer
+
+    def get_queryset(self):
+        valid_subscriptions = True if self.request.query_params.get('valid') == 'true' else False
+        user_id = self.request.query_params.get('user', self.request.user.id)
+        if valid_subscriptions:
+            return UserSubscription.get_valid_subscription(user_id=user_id)
+
+        return UserSubscription.objects.filter(user_id=self.request.user.id)
```

Comment:
What is the case of non valid `UserSubscription`? Can you share some example?

Reply:
A subscription that is active but one can't use it to buy any new course seats. for example a subscription with course limit 5 and expires in 2 months. if a user enrolls in 5 courses within one month, the subscription is still active but is not valid meaning one can't use it to buy any new course. That's the major difference between active and valid subscription